src/wallet/rpcwallet.cpp

```diff
@@ -3796,6 +3795,35 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
         }
     };
 
+    auto push_orchard_result = [&](const libzcash::OrchardRawAddress &addr) -> void {
+        bool hasSpendingKey = pwalletMain->HaveOrchardSpendingKeyForAddress(addr);
+
+        for (const OrchardNoteMetadata& entry: orchardEntries) {
+            auto op = entry.GetOutPoint();
+
+            UniValue obj(UniValue::VOBJ);
+            obj.pushKV("pool", "orchard");
+            obj.pushKV("txid", op.hash.ToString());
+            obj.pushKV("amount", ValueFromAmount(entry.GetNoteValue()));
+            obj.pushKV("amountZat", CAmount(entry.GetNoteValue()));
+            obj.pushKV("memo", HexStr(entry.GetMemo()));
+            obj.pushKV("outindex", (int)op.n);
+            obj.pushKV("confirmations", entry.GetConfirmations());
+
+            txblock BlockData(op.hash);
+            obj.pushKV("blockheight", BlockData.height);
+            obj.pushKV("blockindex", BlockData.index);
+            obj.pushKV("blocktime", BlockData.time);
+
+            if (hasSpendingKey) {
+                bool isInternal = pwalletMain->IsInternalRecipient(addr);
+                obj.pushKV("change", isInternal);
+            }
```

Comment:
This is a scenario that should never arise, because we never construct a UA with an internal receiver, but this is fine.